pygmt/src/rose.py

```diff
@@ -123,21 +124,22 @@ def rose(self, data=None, length=None, azimuth=None, **kwargs):
 
     pen : str
         Set pen attributes for sector outline or rose plot, e.g.
-        ``pen = '0.5p'``. [Default is no outline]. To change pen used to
+        ``pen="0.5p"``. [Default is no outline]. To change pen used to
         draw vector (requires ``vectors``) [Default is same as sector
-        outline] use e.g. ``pen = 'v0.5p'``.
+        outline] use e.g. ``pen="v0.5p"``.
 
     labels : str
```

Comment:
```suggestion
    labels : list of str
```

Reply:
It is (currently) not possible to pass a list to the `labels` parameter of `pygmt.Figure.rose` (see example below). My change in https://github.com/GenericMappingTools/pygmt/pull/2090/commits/08e0b2700fc4ee00272600683666a4f2f6eb8342 is misleading / wrong. I am sorry for this! I adjusted this in https://github.com/GenericMappingTools/pygmt/pull/2090/commits/45df04a80fd7f627fce08241484a03f1d1e31cdf and https://github.com/GenericMappingTools/pygmt/pull/2090/commits/67e00c5f6bc2a612528a38eafeb873a157883de8.

--------------
**Script to reproduce the issue:**
```python

# PyGMT gallery example "Rose diagram"
# >>> without comments; modified <<<
# last access 2022/09/02
# source: https://www.pygmt.org/dev/gallery/histograms/rose.html#sphx-glr-gallery-histograms-rose-py

import pygmt

data = pygmt.datasets.load_sample_data(name="fractures")

fig = pygmt.Figure()

fig.rose(
    length=data.length,
    azimuth=data.azimuth,
    region=[0, 1, 0, 360],
    diameter="7.5c",
    sector="10",
    norm=True,
    color="red3",
    frame=["x0.2g0.2", "y30g30", "+glightgray"],
    pen="1p",
    labels = ["W","N","E","S"], # list, works NOT
    # labels = "W,N,E,S", # string, works
    # labels = ["W,N,E,S"] # works
    # labels = "[W,N,E,S]", # works
    )

fig.show()
```

**Full error message (for list passed through `labels` parameter):**
```
rose [ERROR]: Option -L: Expected
	-L<westlabel>/<eastlabel>/<southlabel>/<northlabel>
rose [ERROR]: Option -L: Given more than once (offending option is -LN)
rose [ERROR]: Option -L: Expected
	-L<westlabel>/<eastlabel>/<southlabel>/<northlabel>
rose [ERROR]: Option -L: Given more than once (offending option is -LS)
rose [ERROR]: Option -L: Expected
	-L<westlabel>/<eastlabel>/<southlabel>/<northlabel>
rose [ERROR]: Option -L: Given more than once (offending option is -LW)
rose [ERROR]: Option -L: Expected
	-L<westlabel>/<eastlabel>/<southlabel>/<northlabel>
Traceback (most recent call last):

  File "C:\ProgramData\Anaconda3\envs\pygmt_env_070_640\lib\site-packages\spyder_kernels\py3compat.py", line 356, in compat_exec
    exec(code, globals, locals)

  File "c:\users\admin\c2\eigenedokumente\studium\promotion\e_gmt\00_testing\general_testing\general_testing.py", line 13, in <module>
    fig.rose(

  File "C:\ProgramData\Anaconda3\envs\pygmt_env_070_640\lib\site-packages\pygmt\helpers\decorators.py", line 585, in new_module
    return module_func(*args, **kwargs)

  File "C:\ProgramData\Anaconda3\envs\pygmt_env_070_640\lib\site-packages\pygmt\helpers\decorators.py", line 725, in new_module
    return module_func(*args, **kwargs)

  File "C:\ProgramData\Anaconda3\envs\pygmt_env_070_640\lib\site-packages\pygmt\src\rose.py", line 207, in rose
    lib.call_module(module="rose", args=build_arg_string(kwargs, infile=fname))

  File "C:\ProgramData\Anaconda3\envs\pygmt_env_070_640\lib\site-packages\pygmt\clib\session.py", line 506, in call_module
    raise GMTCLibError(

GMTCLibError: Module 'rose' failed with status code 72:
rose [ERROR]: Option -L: Expected
	-L<westlabel>/<eastlabel>/<southlabel>/<northlabel>
rose [ERROR]: Option -L: Given more than once (offending option is -LN)
rose [ERROR]: Option -L: Expected
	-L<westlabel>/<eastlabel>/<southlabel>/<northlabel>
rose [ERROR]: Option -L: Given more than once (offending option is -LS)
rose [ERROR]: Option -L: Expected
	-L<westlabel>/<eastlabel>/<southlabel>/<northlabel>
rose [ERROR]: Option -L: Given more than once (offending option is -LW)
rose [ERROR]: Option -L: Expected
	-L<westlabel>/<eastlabel>/<southlabel>/<northlabel>
```


**Output of `python -c "import pygmt; pygmt.show_versions()"`:**
```
PyGMT information:
  version: v0.7.0
System information:
  python: 3.9.13 | packaged by conda-forge | (main, May 27 2022, 16:50:36) [MSC v.1929 64 bit (AMD64)]
  executable: C:\ProgramData\Anaconda3\envs\pygmt_env_070_640\python.exe
  machine: Windows-10-10.0.19043-SP0
Dependency information:
  numpy: 1.23.1
  pandas: 1.4.3
  xarray: 2022.6.0
  netCDF4: 1.6.0
  packaging: 21.3
  geopandas: None
  ghostscript: 9.54.0
  gmt: 6.4.0
GMT library information:
  binary dir: C:/ProgramData/Anaconda3/envs/pygmt_env_070_640
  cores: 4
  grid layout: rows
  library path: C:/ProgramData/Anaconda3/envs/pygmt_env_070_640/Library/bin/gmt.dll
  padding: 2
  plugin dir: C:/ProgramData/Anaconda3/envs/pygmt_env_070_640/Library/bin/gmt_plugins
  share dir: C:/Program Files (x86)/gmt6/share
  version: 6.4.0
```

Reply:
I see. I think passing a list of labels is more Pythonic, but it should be done in a separate PR.